owncloudApp/src/main/java/com/owncloud/android/usecases/transfers/uploads/RetryUploadFromContentUriUseCase.kt

```diff
@@ -16,24 +18,26 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
+
 package com.owncloud.android.usecases.transfers.uploads
 
 import android.content.Context
 import androidx.core.net.toUri
 import androidx.work.WorkManager
-import com.owncloud.android.datamodel.UploadsStorageManager
 import com.owncloud.android.domain.BaseUseCase
 import com.owncloud.android.domain.camerauploads.model.UploadBehavior
+import com.owncloud.android.domain.transfers.TransferRepository
+import com.owncloud.android.domain.transfers.model.TransferStatus
 
 class RetryUploadFromContentUriUseCase(
     private val context: Context,
+    private val transferRepository: TransferRepository,
 ) : BaseUseCase<Unit, RetryUploadFromContentUriUseCase.Params>() {
 
     override fun run(params: Params) {
 
-        val uploadsStorageManager = UploadsStorageManager(context.contentResolver)
-        val failedUploads = uploadsStorageManager.failedUploads
-        val filteredUploads = failedUploads.filter { it.uploadId == params.uploadIdInStorageManager }
+        val failedUploads = transferRepository.getFailedTransfers()
+        val filteredUploads = failedUploads.filter { it.id == params.uploadIdInStorageManager }
```

Comment:
Same as the other retry, we can retrieve directly the upload by its Id, don't you think?

Reply:
Yes! Done